localstack/utils/aws/aws_stack.py

```diff
@@ -220,7 +220,7 @@ def get_local_region():
     if LOCAL_REGION is None:
         session = boto3.session.Session()
         LOCAL_REGION = session.region_name or ""
-    return config.DEFAULT_REGION or LOCAL_REGION
+    return LOCAL_REGION or config.DEFAULT_REGION
```

Comment:
Thanks @calvernaz ! This change effectively causes the region to be returned from `$AWS_DEFAULT_REGION` (an AWS env config), as opposed to the `$DEFAULT_REGION` (our (deprecated) env config). In fact, due to a recent change where we introduced multi-region support for Community, the whole concept of "local regions" or "default regions" is actually deprecated/obsolete. (In other words, at some point we should probably get rid of this entire `get_local_region(..)` function, or at least not depend on it, only as a fallback if no other region can be determined.)

I think the more general issue here is that, for an incoming API Gateway request to, say `apiId123`, we need to identify the target region first. We used to use a cache (lookup table) in `API_REGIONS` (populated upon creation of an API), but this seems like a brittle approach. The more robust approach would be to simply loop through all regions and APIs to find and look up the API with ID `apiId123` (probably acceptable from a performance PoV, given that number of API Gateways is usually low).

Thoughts?

Reply:
Agreed, as I mentioned above it shouldn't get this far in the fallback chain. And since this has already been swapped in the past I'll cancel this.